stdlib/LinearAlgebra/src/adjtrans.jl

```diff
@@ -8,13 +8,57 @@ import Base: length, size, axes, IndexStyle, getindex, setindex!, parent, vec, c
 # note that Adjoint and Transpose must be able to wrap not only vectors and matrices
 # but also factorizations, rotations, and other linear algebra objects, including
 # user-defined such objects. so do not restrict the wrapped type.
+"""
+    Adjoint(A)
+
+Constructs an `Adjoint` view of `A`.
```

Comment:
These seems more like docstrings for `adjoint` and `transpose`. I suggest that we try to document the type instead, i.e.

> Lazy wrapper type for an adjoint view of an `AbstractVector`/`AbstractMatrix`. Usually the `Adjoint` constructor should not be called directly, instead use `adjoint`. To materialize the view use `copy`.

or something. Thoughts? I think we can keep the same example, and the note about `permutedims` is also good.

Reply:
Yes, I agree. I couldn't find a similar example docstring where the type only was documented, usually (e.g., `Symmetric` et al.) emphasize the constructor. I wasn't sure about the `AbstractVector` or matrix, since the comments mention that wrapped type must not be restricted, that's why I kept this aspect completely unmentioned. So, are these really the most abstract types that `Adjoint` would accept?

Reply:
Maybe write something like

> ... view of the underlying object, usually a `AbstractVector`/`AbstractMatrix.